src/bin/ion/commands/beta/generate/model.rs

```diff
@@ -0,0 +1,154 @@
+use ion_schema::isl::isl_type::IslType;
+use std::collections::HashMap;
+use std::ops::Deref;
+// This module contains a data model that code generator cna use to render a template based on the type of the model.
+// Currently, this same data model represented is by `AbstractDataType` but it doesn't hold all the information for the template.
+// e.g. currently there are different fields in template that hold this information like fields, target_kind_name, abstract_data_type.
+// Also, current approach doesn't allow having nested sequences in the generated code. Because the `element_type` in `AbstractDataType::Sequence`
+// doesn't have information on its nested types' `element_type`. This can be resolved with below defined new data model.
+// _Note: This model will eventually use a map (FullQualifiedTypeReference, DataModel) to resolve some the references in container types(sequence or structure)._
+// TODO: This is not yet used in the implementation, modify current implementation to use this data model.
+use crate::commands::beta::generate::context::SequenceType;
+use serde::Serialize;
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, PartialEq)]
+pub struct DataModel {
+    // Represents the fully qualified name for this data model
+    name: FullyQualifiedTypeName,
+    // Represents doc comment for the generated code
+    doc_comment: String,
+    // Represents the type of the data model
+    // It can be `None` for modules or packages.
+    code_gen_type: Option<Box<CodeGenType>>,
+    // Represents the nested types for this data model
+    nested_types: Vec<DataModel>,
+    // Represents the source ISL type which can be used to get other constraints useful for this type.
+    // For example, getting the length of this sequence from `container_length` constraint or getting a `regex` value for string type.
+    // This will also be useful for `text` type to verify if this is a `string` or `symbol`.
+    source: Option<IslType>,
+}
+
+impl DataModel {
+    pub fn is_scalar(&self) -> bool {
```

Comment:
You can use `#![allow(dead_code)]` to suppress warnings about unused methods. Also in the `generate/utils.rs` file:

![image](https://github.com/amazon-ion/ion-cli/assets/611616/8b87b2f8-8730-4f15-8f03-35b937a50a58)